egs2/TEMPLATE/asr1/pyscripts/feats/dump_km_label.py

```diff
@@ -88,16 +106,64 @@ def __call__(self, x):
             return np.argmin(dist, axis=1)
 
 
-def dump_label(rspecifier, in_filetype, wspecifier, out_filetype, km_path, use_gpu):
+def dump_label(
+    rspecifier,
+    in_filetype,
+    wspecifier,
+    out_filetype,
+    km_path,
+    use_gpu,
+    online_feature_extract,
+    **kwargs
+):
+    if online_feature_extract:
+        assert "feature_conf" in kwargs
+        # need to wrap arguments with double-quotes for json string
+        feature_conf = format_feature_conf_str(kwargs["feature_conf"])
+    else:
+        feature_conf = None
+
     apply_kmeans = ApplyKmeans(km_path, use_gpu=use_gpu)
 
-    with file_writer_helper(
-        wspecifier,
-        filetype=out_filetype,
-    ) as writer:
-        for utt, feat in file_reader_helper(rspecifier, in_filetype):
-            lab = apply_kmeans(feat)
-            writer[utt] = lab
+    if not online_feature_extract:
+        # dumped ssl feature in kaldi ark format
+        with file_writer_helper(
+            wspecifier,
+            filetype=out_filetype,
+        ) as writer:
+            for utt, feat in file_reader_helper(rspecifier, in_filetype):
+                lab = apply_kmeans(feat)
+                writer[utt] = lab
+    else:
+        assert feature_conf["type"] in feature_reader_choice
+        reader_class = feature_reader_choice[feature_conf["type"]]
+        reader_conf = feature_conf.get("conf", dict())
+
+        if reader_conf.get("multilayer_feature", None) and isinstance(
+            reader_conf["multilayer_feature"], str
```

Comment:
Maybe we don't need to check whether `reader_conf["multilayer_feature"]` is a `str` or already `bool`, we can directly convert it to `bool`. This will make the code simpler and clearer.
Same to the following `reader_conf["layer"]`.